lib/Model/index.ts

```diff
@@ -163,23 +163,25 @@ function waitForActive (model: Model<DocumentCarrier>, forceRefreshOnFirstAttemp
 	return (): Promise<void> => new Promise((resolve, reject) => {
 		const start = Date.now();
 		async function check (count: number): Promise<void> {
-			try {
-				// Normally we'd want to do `dynamodb.waitFor` here, but since it doesn't work with tables that are being updated we can't use it in this case
-				const tableDetails = (await getTableDetails(model, {"forceRefresh": forceRefreshOnFirstAttempt === true ? forceRefreshOnFirstAttempt : count > 0})).Table;
-				if (tableDetails.TableStatus === "ACTIVE" && (tableDetails.GlobalSecondaryIndexes ?? []).every((val) => val.IndexStatus === "ACTIVE")) {
-					return resolve();
+			if (typeof model.options.waitForActive !== "boolean") {
```

Comment:
If this is just a TypeScript type error, I'm not sure these code level changes are needed.

Reply:
This part was just to make TypeScript happy, because if I remove it then I get these errors:

```
lib/Model/index.ts:177:33 - error TS2339: Property 'check' does not exist on type 'false | ModelWaitForActiveSettings'.
  Property 'check' does not exist on type 'false'.

177                     model.options.waitForActive.check.frequency === 0 ? await utils.set_immediate_promise() : await utils.timeout(model.options.waitForActive.check.frequency);
                                                    ~~~~~

lib/Model/index.ts:177:143 - error TS2339: Property 'check' does not exist on type 'false | ModelWaitForActiveSettings'.
  Property 'check' does not exist on type 'false'.

177                     model.options.waitForActive.check.frequency === 0 ? await utils.set_immediate_promise() : await utils.timeout(model.options.waitForActive.check.frequency);
                                                                                                                                                                  ~~~~~

lib/Model/index.ts:179:58 - error TS2339: Property 'check' does not exist on type 'false | ModelWaitForActiveSettings'.
  Property 'check' does not exist on type 'false'.

179             if (Date.now() - start >= model.options.waitForActive.check.timeout) {
                                                                      ~~~~~                                                                    
```

Reply:
@rajivshah3 Looks like this actually isn't correct. Because if `model.options.waitForActive` is a boolean (like this PR is implementing), nothing will happen. We don't have any else condition here to handle this.

Reply:
Put up a PR for this, but it's causing errors with other tests: #1271. No idea why.

Reply:
Ah yikes, sorry about that. I think I may have made this change under the impression that `waitForActive` would either be `false` or a `object` since that was the inference I made from the docs at that time.

Reply:
@rajivshah3 No problem. Mind reviewing/testing that PR I linked? It should be good to go once Coveralls comes back online from their downtime.